### PR TITLE
Fix: Use unique type for GCPPostgresConnectionConfig

### PR DESCRIPTION
**Pull request by crericha**
Also adds config validation and documentation for GCPPostgresConnectionConfig.

### CODE REVIEW

File: sqlmesh/core/config/connection.py [hunk 1]
```diff
@@ -440,9 +440,29 @@ class GCPPostgresConnectionConfig(_ConnectionConfig):
     db: str
 
     driver: str = "pg8000"
-    type_: Literal["postgres"] = Field(alias="type", default="postgres")
+    type_: Literal["gcp_postgres"] = Field(alias="type", default="gcp_postgres")
```
Comment (eakmanrq) [inline sqlmesh/core/config/connection.py:443]: Ah missed that

File: docs/integrations/engines.md [hunk 2]
```diff
@@ -258,6 +258,18 @@ sqlmesh_airflow = SQLMeshAirflow(
...
 
+## GCP Postgres
+### GCP Postgres - Local/Built-in Scheduler
+**Engine Adapter Type**: `postgres`
+
+| Option                    | Description                                                             |  Type   | Required |
+|---------------------------|-------------------------------------------------------------------------|:-------:|:--------:|
+| `instance_connection_str` | Connection name for the postgres instance                               | string  |    Y     |
+| `user`                    | The username (posgres or IAM) to use for authentication                 | string  |    Y     |
+| `password`                | The password to use for authentication when using a Postgres user       | string  |    N     |
```
Comment (eakmanrq) [inline docs/integrations/engines.md:269]: Can you add a comment that says it is required if not using iam?

File: docs/integrations/engines.md [hunk 3]
```diff
@@ -258,6 +258,18 @@ sqlmesh_airflow = SQLMeshAirflow(
...
+| `enable_iam_auth`         | Enables IAM authentication when using a IAM user                        | boolean |    N     |
```
Comment (eakmanrq) [inline docs/integrations/engines.md:270]: Can you add a comment saying it is required if not using password?

File: sqlmesh/core/config/connection.py (hunk 1 above)
Comment (crericha) [reply sqlmesh/core/config/connection.py:443]: Iaroslav caught it :-)

**Review by izeigerman — APPROVED** [review]

**Pull request merged**